src/main/daml/Daml/Finance/Settlement/Batch.daml

```diff
@@ -57,28 +61,26 @@ template Batch
         let
           settleInstruction instructionKey = do
             let context = show instructionKey
+            -- execute instruction and get used holding standards for the instrument
             (instructionCid, instruction) <- fetchByKey @Instruction instructionKey
-            pledged <- case instruction.allocation of
-              Pledge cid -> Some <$> fetch cid
-              _ -> pure None
+            pledgedHoldingStandard <- case instruction.allocation of
+              Pledge cid -> L.singleton <$> getHoldingStandard cid
+              _ -> pure []
             settledCid <- exercise (toInterfaceContractId @Instruction.I instructionCid)
               Instruction.Execute with actors = actors `S.union` requestors
-            settled <- T.mapA fetch settledCid
+            settledHoldingStandard <- optionalToList <$> T.mapA getHoldingStandard settledCid
+
             let
-              usedInterfaceTypeReps =
-                fmap ((instruction.routedStep.quantity.unit, ) . interfaceTypeRep) <$>
-                [pledged, settled]
-            settledCid <-
-              T.mapA
-                (\cid -> fromOptional cid <$> undisclose (context, settlers) actors cid)
-                settledCid
-            pure (settledCid, usedInterfaceTypeReps)
+              instrumentWithHoldingStandards = fmap ((instruction.routedStep.quantity.unit, )) $
```

Comment:
also, I think you can use `<$>` instead of `$` and avoid having to write `fmap`

Reply:
changed it already